pkg/diagnostics/diagnostics.go

```diff
@@ -0,0 +1,121 @@
+package diagnostics
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zbus"
+	"github.com/threefoldtech/zos/pkg/perf"
+	"github.com/threefoldtech/zos/pkg/utils"
+)
+
+const (
+	callTimeout    = 3 * time.Second
+	testNetworkKey = "perf.network-health"
+)
+
+// Modules is all the registered modules on zbus
+var Modules = []string{
+	"storage",
+	"node",
+	"identityd",
+	"vmd",
+	"flist",
+	"network",
+	"container",
+	"provision",
+	"gateway",
+	"qsfsd",
+}
+
+type moduleStatus struct {
+	Status zbus.Status `json:"status,omitempty"`
+	Err    error       `json:"error,omitempty"`
+}
+
+// Diagnostics show the health of zbus modules
+type Diagnostics struct {
+	// SystemStatusOk is the overall system status
+	SystemStatusOk bool `json:"system_status_ok"`
+	// ZosModules is a list of modules with their objects and workers
+	ZosModules map[string]moduleStatus `json:"modules"`
+	// Online is the state of the grid services reachable from the node
+	Online bool `json:"online"`
+}
+
+func GetSystemDiagnostics(ctx context.Context, msgBrokerCon string) (Diagnostics, error) {
```

Comment:
it's really bad idea to pass around the msgBrokerCon string instead of the zbus.Client itself. the zbus.Client can be used concurrently by multiple stubs with no issue. Creating a Client creates a pool that is kept alive forever. Hence  creating a client each time means you create a new pool each time u call this function!